### DISCUSSION TITLE
Unable to resolve "@socket.io/component-emitter" from "node_modules\socket.io-client\build\esm\manager.js"  React Native Expo

### DISCUSSION THREAD

**Discussion by Rise-admin1** [socketio/socket.io#4930] [Q&A]
I installed socket.io on the client side using ```npm install socket.io-client``` and created a socket file 
```
`import { io } from "socket.io-client";
import {Platform} from 'react-native';


export const BaseUrl  = () => Platform.OS === 'android' ? 'http://10.0.2.2:8081' : 'http://localhost:8081'; 

export const socket = io.connect("http://10.0.2.2:3000"); 
```
When running the metro builder, I get an error ```Unable to resolve "@socket.io/component-emitter" from "node_modules\socket.io-client\build\esm\manager.js"```

What is a possible fix?

**Comment by Rise-admin1** [comment]
Works with 4.1.2 though.

**Comment by darrachequesne** [comment]
That's really weird.

In version 4.1.2, the package was named `component-emitter`: https://github.com/socketio/socket.io-client/blob/b466c6f0b1dd5ef1a2149abc89d1884d376c0eee/package.json#L32C1-L32C35

In latest version, we have forked our own version to `@socket.io/component-emitter`: https://github.com/socketio/socket.io-client/blob/8cfea8c30b113b0b6987976af9243cba6f537f30/package.json#L48

Not sure why the metro builder complains though. Maybe the fact that it's a scoped package ("@socket.io/...")?
